src/bindings/c/include/openvino/c/ov_compiled_model.h

```diff
@@ -139,25 +139,28 @@ OPENVINO_C_API(ov_status_e)
 ov_compiled_model_create_infer_request(const ov_compiled_model_t* compiled_model, ov_infer_request_t** infer_request);
 
 /**
- * @brief Sets properties for the current compiled model.
+ * @brief Sets properties for a device, acceptable keys can be found in ov_property_key_xxx.
  * @ingroup compiled_model
  * @param compiled_model A pointer to the ov_compiled_model_t.
- * @param property ov_properties_t.
+ * @param property_key The property key.
+ * @param ... variadic paramaters The format is <ov_property_key, ov_property_key>.
  * @return Status code of the operation: OK(0) for success.
  */
 OPENVINO_C_API(ov_status_e)
-ov_compiled_model_set_property(const ov_compiled_model_t* compiled_model, const ov_properties_t* property);
+ov_compiled_model_set_property(const ov_compiled_model_t* compiled_model, ...);
```

Comment:
@riverlijunjie Why here don't we have a number of properties?
Because in the compiled model we couldn't avoid an additional argument.

Reply:
Here just to keep a simple and similar interface with C++ API, multiple properties we can call this API multiple times.
It is fine for me to add additional argument to keep align with create compiled model.

Reply:
I just would like to understand why the same approach doesn't work for `compile_model()`

Reply:
ov_core/compiled_mode_set_property() always set a property.
while ov_core_compile_model() is used to compile model with or without properties, so to avoid va_arg() to read invalid pointer in case of no property, we should add a property size argument to make it be aware.